haskell/private/actions/compile.bzl

```diff
@@ -93,16 +92,59 @@ def _process_hsc_file(hs, cc, ghc_defs_dump, hsc_file):
     inputs = depset(transitive = [
       depset(cc.hdrs),
       depset([hs.tools.gcc]),
-      depset([hsc_file, ghc_defs_dump])
+      depset([hsc_file, ghc_defs_dump]),
     ]),
-    outputs = [hs_out, hsc_output_dir],
+    outputs = [hs_out],
     mnemonic = "HaskellHsc2hs",
     executable = hs.tools.hsc2hs,
     arguments = [args],
     env = hs.env,
   )
   return hs_out
 
+def _process_chs_file(hs, cc, ghc_defs_dump, chs_file):
+  """Process a single chs file.
+
+  Args:
+    hs: Haskell context.
+    cc: CcInteropInfo, information about C dependencies.
+    ghc_defs_dump: File with GHC definitions.
+    chs_file: chs file to process.
+
+  Returns:
+    File: Haskell source file created by processing chs_file.
+  """
+
+  args = hs.actions.args()
+
+  # Output a Haskell source file.
+  chs_dir_raw = target_unique_name(hs, "chs")
+  hs_out = declare_compiled(hs, chs_file, ".hs", directory=chs_dir_raw)
+  args.add([chs_file.path, "-o", hs_out.path])
+
+  args.add(["-C-E"])
+  args.add(["--cpp", hs.tools.gcc.path])
+  # XXX c2hs doesn't seem to support direct inclusion of arbitrary headers
+  # args.add(["-C-I{0}".format(ghc_defs_dump.dirname)])
+  # args.add(["-C-i{0}".format(ghc_defs_dump.basename)])
```

Comment:
I believe that you can do this with `-C-include{0}`.  (The `-include` flag to gcc should have the same effect as `-i` to `hsc2hs`.)

Reply:
Should be fixed now.